test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/ArchiveConsistency.java

```diff
@@ -68,20 +94,68 @@ private static void doTest(String baseArchiveName, String topArchiveName) throws
             throw new IOException(jsa + " does not exist!");
         }
 
-        // Modify the CRC values in the header of the top archive.
+        // 1. Modify the CRC values in the header of the top archive.
+        System.out.println("\n1. Modify the CRC values in the header of the top archive");
         String modTop = getNewArchiveName("modTopRegionsCrc");
         File copiedJsa = CDSArchiveUtils.copyArchiveFile(jsa, modTop);
         CDSArchiveUtils.modifyAllRegionsCrc(copiedJsa);
 
-        run2(baseArchiveName, modTop,
-            "-Xlog:class+load",
-            "-Xlog:cds+dynamic=debug,cds=debug",
-            "-XX:+VerifySharedSpaces",
-            "-cp", appJar, mainClass)
-            .assertAbnormalExit(output -> {
-                    output.shouldContain("Header checksum verification failed")
-                          .shouldContain("Unable to use shared archive")
-                          .shouldHaveExitValue(1);
-                });
+        runTwo(baseArchiveName, modTop,
+               appJar, mainClass, 1,
+               new String[] {"Header checksum verification failed",
+                             "Unable to use shared archive"});
+
+        // 2. Make header size biger than the archive size
+        System.out.println("\n2. Make header size biger than the archive size");
+        String largerHeaderSize = getNewArchiveName("largerHeaderSize");
+        copiedJsa = CDSArchiveUtils.copyArchiveFile(jsa, largerHeaderSize);
+        CDSArchiveUtils.modifyHeaderIntField(copiedJsa, CDSArchiveUtils.offsetHeaderSize,  (int)copiedJsa.length() + 1024);
+        runTwo(baseArchiveName, largerHeaderSize,
+               appJar, mainClass, 1,
+               new String[] {"_header_size should be equal to _base_archive_path_offset plus _base_archive_name_size",
+                             "Unable to use shared archive"});
+
+        // 3. Make base archive path offset beyond of header size
+        System.out.println("\n3. Make base archive path offset beyond of header size.");
+        String wrongBaseArchivePathOffset = getNewArchiveName("wrongBaseArchivePathOffset");
+        copiedJsa = CDSArchiveUtils.copyArchiveFile(jsa, wrongBaseArchivePathOffset);
+        int fileHeaderSize = (int)CDSArchiveUtils.fileHeaderSize(copiedJsa);
+        int baseArchivePathOffset = CDSArchiveUtils.baseArchivePathOffset(copiedJsa);
+        CDSArchiveUtils.modifyHeaderIntField(copiedJsa, CDSArchiveUtils.offsetBaseArchivePathOffset, baseArchivePathOffset + 1024);
```

Comment:
In line 124, could `baseArchivePathOffset` be used instead of `CDSArchiveUtils.offsetBaseArchivePathOffset` ?